### PR TITLE
Update "rand" to 0.8 and "rand_distr" to 0.4.

### PR DESCRIPTION
**Pull request by ghuls**

### CODE REVIEW

**Comment by ghuls** [comment]
`polars/polars-core/Cargo.toml` was using 0.7 and `polars/Cargo.toml` was using 0.8.

**Comment by ghuls** [comment]
I guess 0.7 was used instead of 0.8 because of this change: `Remove wasm-bindgen and stdweb feature flags. For details of WASM support, see the getrandom documentation. (#948)`

Seems like it is not difficult to compile for webassembly too:
https://docs.rs/getrandom/0.2.3/getrandom/#webassembly-support

**Comment by ritchie46** [comment]
And WASM compilation is still easier on 0.7? I haven't looked into this much yet.,

**Comment by ghuls** [comment]
I didn't try to compile to WASM yet, but it is supposed to be relatively easy:
```
Indirect Dependencies

If getrandom is not a direct dependency of your crate, you can still enable any of the above fallback behaviors by enabling the relevant feature in your root crate’s Cargo.toml:

[dependencies]
getrandom = { version = "0.2", features = ["js"] }
```

@universalmind303 Can you confirm?

**Comment by ritchie46** [comment]
Maybe we should add a WASM compile check in CI first. I don't want to break that support.

**Comment by ghuls** [comment]
WASM compiles with the latest version:
```
cargo build --target wasm32-unknown-unknown --no-default-features --features=random heeft cargo build --target wasm32-unknown-unknown --no-default-features --features=random
```

File: polars/polars-core/Cargo.toml [hunk 1]
```diff
@@ -134,6 +134,7 @@ arrow = { package = "arrow2", git = "https://github.com/jorgecarleitao/arrow2",
 # arrow = { package = "arrow2", version = "0.7", default-features = false, features = ["compute"] }
 chrono = { version = "0.4", optional = true }
 comfy-table = { version = "4.0", optional = true }
+getrandom = { version = "0.2", features = ["js"] }
```
Comment (ritchie46) [inline polars/polars-core/Cargo.toml:137]: This one dhould be added by downstream users right? We don't need it in polars itself. Especially the js target not.

**Comment by ritchie46** [comment]
Tgmhanks @ghuls. Sorry for the confusion on my end.

**Pull request merged**